torch/nn/modules/module.py

```diff
@@ -193,13 +193,48 @@ def _apply(self, fn):
         for module in self.children():
             module._apply(fn)
 
-        for param in self._parameters.values():
+        def compute_should_change_tensor_inplace(tensor, tensor_applied):
```

Comment:
Note to self: rename the compute function to `compute_should_use_data`, and in the function check `is_same_tensorimpl_type`, because `_apply` can be used to e.g. move TensorImpl to QTensorImpl, in which case they are the same device type.